dll/win32/uxtheme/nonclient.c

```diff
@@ -1121,22 +1233,88 @@ HRESULT WINAPI DrawNCPreview(HDC hDC,
     context.scrolltheme = OpenThemeDataFromFile(hThemeFile, hwndDummy, L"SCROLLBAR", 0);
     if (!context.scrolltheme)
         return E_FAIL;
-    context.Active = TRUE;
     context.wi.cbSize = sizeof(context.wi);
     if (!GetWindowInfo(hwndDummy, &context.wi))
         return E_FAIL;
     context.wi.dwStyle |= WS_VISIBLE;
-    context.CaptionHeight = context.wi.cyWindowBorders;
-    context.CaptionHeight += GetSystemMetrics(context.wi.dwExStyle & WS_EX_TOOLWINDOW ? SM_CYSMCAPTION : SM_CYCAPTION );
+
     context.hRgn = CreateRectRgnIndirect(&context.wi.rcWindow);
+    RECT rcAdjPreview = { prcPreview->left, prcPreview->top, prcPreview->right, prcPreview->bottom };
+    INT previewWidth = rcAdjPreview.right - rcAdjPreview.left;
+    INT previewHeight = rcAdjPreview.bottom - rcAdjPreview.top;
 
-    /* Paint the window on the preview hDC */
-    rcCurrent = context.wi.rcWindow;
-    OffsetRect( &rcCurrent, -context.wi.rcWindow.left, -context.wi.rcWindow.top);
-    SetViewportOrgEx(hDC, context.wi.rcWindow.left, context.wi.rcWindow.top, NULL);
-    ThemeCalculateCaptionButtonsPos(hwndDummy, context.theme);
-    ThemePaintWindow(&context, &rcCurrent, FALSE);
-    SetViewportOrgEx(hDC, 0, 0, NULL);
+    /* Draw inactive preview window */
+    context.Active = FALSE;
+    SetWindowTextW(hwndDummy, L"Inactive Window");
+    DrawWindowForNCPreview(hDC, &context, rcAdjPreview.left, rcAdjPreview.top, rcAdjPreview.right - 17, rcAdjPreview.bottom - 20, TRUE, NULL);
+
+    /* Draw active preview window */
+    context.Active = TRUE;
+    SetWindowTextW(hwndDummy, L"Active Window");
+
+    DWORD textDrawFlags = DT_NOPREFIX | DT_SINGLELINE | DT_WORDBREAK;
+    RECT rcWindowClient;
+    DrawWindowForNCPreview(hDC, &context, rcAdjPreview.left + 10, rcAdjPreview.top + 22, rcAdjPreview.right, rcAdjPreview.bottom, TRUE, &rcWindowClient);
+    SetTextColor(hDC, GetThemeSysColor(context.theme, COLOR_WINDOWTEXT));
+    LOGFONTW lfText;
+    HFONT textFont = NULL;
+    if (SUCCEEDED(GetThemeSysFont(context.theme, TMT_MSGBOXFONT, &lfText)))
+        textFont = CreateFontIndirectW(&lfText);
+
+    if (textFont)
+        SelectFont(hDC, textFont);
+
+    DrawThemeText(context.theme, hDC, WP_DIALOG, 0, L"Window Text", -1, DT_LEFT | DT_TOP | textDrawFlags, 0, &rcWindowClient);
+
+    /* Draw preview dialog window */
+    SetWindowTextW(hwndDummy, L"Message Box");
```

Comment:
String may be localized in the future? (And same question for some other ones I see there, above and below)

Reply:
I re-iterate:

> It...should be localized, should it not?